product_variant_inactive/models/product_variant.py

```diff
@@ -44,8 +50,40 @@ def fields_view_get(
         return res
 
     def write(self, vals):
-        if self._context.get("skip_reactivate_variant") and vals == {"active": True}:
-            _logger.info("Skip reactivating product %s" % self.ids)
-            return True
-        else:
-            return super().write(vals)
+        if vals.get("active") and self._context.get("unset_combination_deleted"):
+            vals["combination_deleted"] = False
+        for product in self:
+            if (
+                self._context.get("skip_reactivate_variant")
+                and vals.get("active", False)
+                and not product.active
+                and not product.combination_deleted
+            ):
+                _logger.info("Skip reactivating product %s" % self.ids)
+            else:
+                super(ProductProduct, product).write(vals)
+        return True
+
+    def _unlink_or_archive(self, check_access=True):
+        records = self
+        if not self._context.get("skip_filter_deleted"):
+            records = records.filtered(
+                lambda p: not p.combination_deleted
+            ).with_context(skip_filter_deleted=True)
+            records.write({"combination_deleted": True})
```

Comment:
You should write "active: False" here it will avoid having an hack in the constraint
```suggestion
            records.write({"combination_deleted": True, "active": False})
```